pool.go

```diff
@@ -171,23 +171,59 @@ func (p *connPool) new() (*conn, error) {
 		)
 		return nil, err
 	}
+
 	cn, err := p.dial()
 	if err != nil {
 		p.lastDialErr = err
+		return nil, err
 	}
-	return cn, err
+
+	if err := p.init(cn); err != nil {
+		p.Remove(cn)
+		return nil, err
+	}
+
+	return cn, nil
+}
+
+// Initialize connection
+func (p *connPool) init(cn *conn) error {
```

Comment:
Please rename to `initConn`. Just `init` should initialize pool, not connection.